shared/teams/team/member/add-to-channels.tsx

```diff
@@ -3,53 +3,95 @@ import * as Kb from '../../../common-adapters'
+  }, [dispatch, teamID])
 
-const AddToChannels = ({teamID, username}: Props) => {
   const meta = Container.useSelector(s => Constants.getTeamMeta(s, teamID))
-  const channelInfos = Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  const channelInfos = getChannelsForList(
+    Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  )
   const [filter, setFilter] = React.useState('')
   const filterLCase = filter.trim().toLowerCase()
   const [filtering, setFiltering] = React.useState(false)
   const channels = filterLCase
-    ? [...channelInfos.values()].filter(c => c.channelname.toLowerCase().includes(filterLCase))
-    : [...channelInfos.values()]
+    ? channelInfos.filter(c => c.channelname.toLowerCase().includes(filterLCase))
+    : channelInfos
   const items = [
     ...(filtering ? [] : [{type: 'header' as const}]),
     ...channels.map(c => ({
       channelname: c.channelname,
+      conversationIDKey: c.conversationIDKey,
       numMembers: c.numParticipants,
       type: 'channel' as const,
     })),
   ]
-  const [selected, setSelected] = React.useState(new Set<string>())
-  const onSelect = (channelname: string) => {
-    if (selected.has(channelname)) {
-      selected.delete(channelname)
+  const [selected, setSelected] = React.useState(new Set<ChatTypes.ConversationIDKey>())
+  const onSelect = (convIDKey: ChatTypes.ConversationIDKey) => {
+    if (selected.has(convIDKey)) {
+      selected.delete(convIDKey)
       setSelected(new Set(selected))
     } else {
-      selected.add(channelname)
+      selected.add(convIDKey)
       setSelected(new Set(selected))
     }
   }
-  const onSelectAll = () => setSelected(new Set([...channelInfos.values()].map(c => c.channelname)))
+  const onSelectAll = () => setSelected(new Set([...channelInfos.values()].map(c => c.conversationIDKey)))
   const onSelectNone = () => setSelected(new Set())
 
-  const onCancel = () => {} // TODO
-  const onFinish = () => {} // TODO useRPC probably
+  const onCancel = () => dispatch(nav.safeNavigateUpPayload())
+
+  const submit = Container.useRPC(RPCChatGen.localBulkAddToManyConvsRpcPromise)
+  const [waiting, setWaiting] = React.useState(false)
+  const onFinish = () => {
+    if (!selected.size) {
+      onCancel()
+      return
+    }
+    setWaiting(true)
+    submit(
+      [{conversations: [...selected].map(ChatTypes.keyToConversationID), usernames}],
+      () => {
+        setWaiting(false)
+        onCancel()
+      },
+      error => {
+        console.error(error)
+        setWaiting(false)
+      }
+    )
+  }
+
   const numSelected = selected.size
 
   const renderItem = (_, item: Unpacked<typeof items>) => {
     switch (item.type) {
       case 'header': {
-        const allSelected = selected.size === channelInfos.size
+        const allSelected = selected.size === channelInfos.length
         return (
           <HeaderRow
             key="{header}"
```

Comment:
is there a separate ticket for hooking up the `onCreate`?

Reply:
I'll add the nav